### PR TITLE
Integrate goreleaser to produce release images

### PR DESCRIPTION
**Pull request by timflannagan**
Integrate goreleaser into this repository for building and pushing release container images (and multi-arch manifests).

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - amd64
+    ldflags:
+    - -X {{ .Env.PKG }}.GitCommit={{ .ShortCommit }}
+  - id: plain-v0
+    main: ./provisioner/plain-v0
+    binary: plain-v0
+    goos:
+    - linux
+    goarch:
+    - amd64
```
Comment (timflannagan) [inline .goreleaser.yml:27]: note: linux-amd64 is the only supported container image right now. I just wanted to get consensus on this before I expand the list of archs (follow-up?).

**Comment by timflannagan** [comment]
I was playing around with this in my fork. You can see that https://quay.io/repository/tflannag/plain-v0-provisioner?tab=tags&tag=latest got populated with a new manifest + individual platform tag.

File: .goreleaser.yml (hunk 1 above)
Comment (tylerslaton) [reply .goreleaser.yml:27]: I'd say a follow-up would make sense as we're already adding a lot here.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: release
+
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - 'v*'
+  pull_request:
```
Comment (tylerslaton) [inline .github/workflows/release.yaml:9]: Do we want this to run on pull requests - wouldn't we only release on a tag event? Also, I notice that we are specifying on `push` events but the only thing those trigger is a `checkout`, `install` and `login`.

**Review by tylerslaton — COMMENTED** [review]
Looks good to me. Just want to iron out some details around what triggers the workflow.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (timflannagan) [reply .github/workflows/release.yaml:9]: We're running goreleaser on every PR and merge to any upstream branch right now, but updating the goreleaser arguments and logging into quay when a tag event has been detected. Does that make sense? We could also release rukpak when changes land in the main branch too.

File: .goreleaser.yml (hunk 1 above)
Comment (timflannagan) [reply .goreleaser.yml:27]: I think I'll look into combing the deploy + release workflow in a follow-up PRs as well, so both of them build and publish containers through goreleaser.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      if: startsWith(github.ref, 'refs/tags')
+      with:
+        registry: quay.io
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+
+    - name: Run goreleaser
+      run: |
+        if [[ $GITHUB_REF == refs/tags/* ]]; then
+          export GORELEASER_ARGS="--rm-dist"
```
Comment (exdx) [inline .github/workflows/release.yaml:39]: It looks like we pass the `--rm-dist` argument in the Makefile as well -- should we be passing it twice?

**Review by exdx — APPROVED** [review]
one small nit

File: .github/workflows/release.yaml (hunk 3 above)
Comment (timflannagan) [reply .github/workflows/release.yaml:39]: So we're actually overriding the default GORELEASER_ARGS variable, and _removing_ the --snapshot option. It's not the most readable (or obvious) implementation, so I can understand any potential confusion.

File: .goreleaser.yml (hunk 1 above)
Comment (tylerslaton) [reply .goreleaser.yml:27]: Yeah that sounds like a great idea.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (tylerslaton) [reply .github/workflows/release.yaml:9]: So we're running goreleaser on all of these events, but it only actually releases on a tag event?

**Review by tylerslaton — APPROVED** [review]
Looks good to me, good to see the release automation come through.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (timflannagan) [reply .github/workflows/release.yaml:9]: Yup that's the correct read.

**Pull request merged**